### DISCUSSION TITLE
How to use motion control on DX100

### DISCUSSION THREAD

**Discussion by ljhsgdsb** [Yaskawa-Global/YMConnect#12] [Q&A] [answered]
Hello,
I have already  build a "Hello World" app using YMConnect and the project runs successfully on DX100.
I want to use YMConnect to plan the trajectory of the robot. Can I input a set of Cartesian points and have the robot fit a smooth curve to those points and move along that curve at a given speed? Or do I have to input the joint coordinates of a series of points?
How should I code if I can, because in the development documentation "MotionManagerInterface" is not available on DX100 controller.
Thank you very much for your answer.

**Comment by ted-miller** [comment]
Hi @ljhsgdsb
Direct motion control is not available through YMConnect for the DX100. 

You could use YMConnect to set some Position (P) variables that are referenced in an INFORM job. Then you could remotely start/stop that job.

**Reply by ljhsgdsb** [reply]
Hi,
Do you have a standardized procedure for configuring Position(p) variables and remotely initiating and terminating jobs?
The provision of pertinent information would be greatly appreciated.

**Reply by ted-miller** [reply, marked as answer]
There is no standard method for this type of behavior. It is highly dependent on the specific application for your particular robot.

Generally speaking:

- Create some job on the pendant with something like `MOVL P001`
- Enable servo power: https://developer.motoman.com/en/YMConnect/ControlCommandsInterface#setservos
- Write P variables: https://developer.motoman.com/en/YMConnect/VariableInterface#robotpositionvariable
- Set active job: https://developer.motoman.com/en/YMConnect/JobInterface#setactivejob
- Start job: https://developer.motoman.com/en/YMConnect/ControlCommandsInterface#startjob

**Comment by ted-miller** [comment]
Closing this discussion. Feel free to reopen if there are additional questions.
